src/game/RandomizationOfShipsPositions.js

```diff
@@ -0,0 +1,163 @@
+import { SHIPS_LIST } from "./constants";
+
+const GAME_BOARD = document.querySelectorAll(".board__fields");
+
+let gameBoardArray = [];
+let fieldsNamesArr = [];
+let direction;
+let position;
+const GAME_BOARD_WIDTH = 10;
+
+const createFieldsName = () => {
+  for (let i = 0; i < 10; i++) {
+    for (let j = 0; j < 10; j++) {
+      fieldsNamesArr.push(`${String.fromCharCode(i + 65)}${j + 1}`);
+    }
+  }
+};
+createFieldsName();
+
+const createGameBoardArray = () => {
+  gameBoardArray = [];
+  for (let i = 0; i < GAME_BOARD_WIDTH; i++) {
+    gameBoardArray.push([]);
+    for (let j = 0; j < GAME_BOARD_WIDTH; j++) {
+      gameBoardArray[i].push({
+        isEmpty: true,
+        shipName: "",
+      });
+    }
+  }
+};
+
+const directionRoll = () => {
+  Math.floor(Math.random() * 2) === 0
+    ? (direction = "vertical")
+    : (direction = "horizontal");
+};
+
+const makePosition = (shipIndex) => {
+  position = [];
+  for (let i = 0; i < 2; i++) {
+    position.push(Math.floor(Math.random() * GAME_BOARD_WIDTH));
+  }
+  if (direction === "horizontal") {
+    if (
+      Number(`${position[1]}`) + SHIPS_LIST[shipIndex].size >
+      GAME_BOARD_WIDTH
+    ) {
+      position[1] = Math.abs(
+        Number(`${position[1]}`) - SHIPS_LIST[shipIndex].size
+      );
+    }
+  } else {
+    if (
+      Number(`${position[0]}`) + SHIPS_LIST[shipIndex].size >
+      GAME_BOARD_WIDTH
+    ) {
+      position[0] = Math.abs(
+        Number(`${position[0]}`) - SHIPS_LIST[shipIndex].size
+      );
+    }
+  }
+};
+
+const addShip = () => {
+  createGameBoardArray();
+  for (let shipIndex = 0; shipIndex < SHIPS_LIST.length; shipIndex++) {
+    directionRoll();
+    if (direction === "horizontal") {
+      horizontalShipPosition(shipIndex);
+    } else {
+      verticalShipPosition(shipIndex);
+    }
+  }
+};
+
+const verticalShipPosition = (shipIndex) => {
+  makePosition(shipIndex);
+  let crashTestArr = [];
+  for (let i = 0; i < SHIPS_LIST[shipIndex].size; i++) {
+    crashTestArr.push(gameBoardArray[position[0] + i][position[1]]);
+  }
+  if (crashTestArr.some((el) => el.isEmpty === false)) {
+    verticalShipPosition(shipIndex);
+  } else {
+    for (let i = 0; i < SHIPS_LIST[shipIndex].size; i++) {
+      gameBoardArray[position[0] + i].fill(
+        {
+          isEmpty: false,
+          shipName: SHIPS_LIST[shipIndex].shipName,
+        },
+        position[1],
+        position[1] + 1
+      );
+    }
+  }
+};
+
+const horizontalShipPosition = (shipIndex) => {
+  makePosition(shipIndex);
+  let crashTestArr = gameBoardArray[position[0]].slice(
```

Comment:
I have a weird feeling this operation could be much simpler. But it works so ... for now, I'm cool with it.